pkg/systemd/hwinfo.jsx

```diff
@@ -24,61 +24,195 @@ import React from "react";
+                            <th>{ _("BIOS date") }</th>
+                            <td>{ moment(info.bios_date).isValid() ? moment(info.bios_date).format('L') : info.bios_date }</td>
+                        </tr>
+                    </React.Fragment>
+                    }
+                    { info.nproc !== undefined &&
+                    <React.Fragment>
+                        <tr>
+                            <th>{ _("CPU") }</th>
+                            <td>{ (info.nproc > 1) ? `${info.nproc}x ${info.cpu_model}` : info.cpu_model }</td>
+                        </tr>
+                        { onSecurityClick !== undefined &&
+                        <tr>
+                            <th>{ _("CPU Security") }</th>
+                            <td>{ (this.state.allowed && <a role="link" onClick={onSecurityClick}>{ _("Mitigations") }</a>) ||
+                                <OverlayTrigger overlay={ <Tooltip id="tip-cpu-security">{ cockpit.format(_("The user $0 is not permitted to change cpu security mitigations"), permission.user ? permission.user.name : '') }</Tooltip> }><span>{ _("Mitigations") }</span></OverlayTrigger> }
+                            </td>
+                        </tr>
+                        }
+                    </React.Fragment>
+                    }
+                </tbody>
+            </table>
+        );
+    }
+}
+
+class CPUSecurityMitigationsDialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.close = this.close.bind(this);
+        this.saveAndReboot = this.saveAndReboot.bind(this);
+        this.state = {
+            nosmt: undefined,
+            alert: undefined,
+        };
+        CPUSecurityMitigationsDialog.availableMitigations().then(({ nosmt_enabled }) => {
+            this.setState({ nosmt: nosmt_enabled });
+        });
+    }
+
+    static availableMitigations() {
+        /* nosmt */
+        let promises = [cockpit.spawn(["lscpu"], { environ: ["LC_ALL=C.UTF-8"], }), cockpit.file("/proc/cmdline").read()];
+        return Promise.all(promises).then(values => {
+            let threads_per_core;
+            try {
+                threads_per_core = Number(values[0].split('\n').filter(l => l.indexOf('Thread(s) per core:') !== -1)[0].split(':')[1]);
+            } catch (e) {
+                console.warn(e);
+                return { available: false };
             }
-        </tbody>
-    </table>
-);
+            /* "nosmt" and "nosmt=force" are valid */
+            let nosmt_enabled = (values[1].indexOf("nosmt") !== -1 && values[1].indexOf("nosmt=") === -1) || values[1].indexOf("nosmt=force") !== -1;
+            /* available if threads>1 and the cmdline is valid */
+            let nosmt_available = (threads_per_core > 1 || (threads_per_core === 1 && nosmt_enabled)) && (values[1].indexOf("nosmt=") === -1 || values[1].indexOf("nosmt=force") !== -1);
+            return { available: nosmt_available, nosmt_enabled: nosmt_enabled };
```

Comment:
Having an `available` state field which is true iff *any* `*_available` is true, is a good thing IMHO, as it can be used to decide whether or not to show the "Mitigations" link. This should be computed at the end, so avoid any `return`s above.

Reply:
definitely, adding the next option this available should be calculated on the basis of the availability of the mitigations.